### PR TITLE
Gatsby plugin local search

### PR DESCRIPTION
**Pull request by ionab**
Introduces search and fixes the IIIF and Github Links

### CODE REVIEW

File: src/components/Search/SearchForm.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      >
+        <label htmlFor="search-input"></label>
+        <button type="submit" style={{ border: 'none', background: 'none' }}>
+          <SearchIcon color="black" />
+        </button>
+        <input
+          type="search"
+          id="search-input"
+          name="keywords"
+          style={{ width: '90%', border: 'none', height: '100%' }}
```
Comment (stephenwf) [inline src/components/Search/SearchForm.js:52]: Few small snags we could improve on this box:
* Text is very small
* Focus is very narrow (and has an outline - image attached)
* Text is a bit too high for the size of box

Screenshot 2020-07-27 at 17 04 33


I don't think you can apply `:focus` in the style, may have to switch this over to a css class name.

File: src/components/Search/SearchForm.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  // };
+
+  // const handleSubmit = e => {
+  //   e.preventDefault();
+  //   navigate(`/${pageLanguage}/search/?keywords=${encodeURIComponent(value)}`);
+  // };
+
+  return showTitle ? (
+    <>
+      {showTitle ? <h1>Search posts</h1> : ''}
```
Comment (stephenwf) [inline src/components/Search/SearchForm.js:33]:
```suggestion
      <h1>Search posts</h1>
```

File: src/components/Search/SearchForm.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    //     name="keywords"
+    //     value={value}
+    //     onChange={e => handleChange(e)}
+    //   />
+    //   <button type="submit">
+    //     <SearchIcon />
+    //   </button>
+    // </form>
+    <button
+      style={{ border: 'none', background: 'none' }}
```
Comment (stephenwf) [inline src/components/Search/SearchForm.js:79]: Search icon and button alignment is a bit off. I found adding `alignSelf: 'flex-end'` to the button and `display: 'block'` to the SVG icon did the trick.

Screenshot 2020-07-27 at 16 54 24

Screenshot 2020-07-27 at 17 15 01

File: src/components/Search/SearchForm.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+          name="keywords"
+          style={{ width: '90%', border: 'none', height: '100%' }}
+          value={query ? query : ''}
+          onChange={e =>
+            navigate(
+              `/${pageLanguage}/search/?keywords=${encodeURIComponent(
+                e.target.value
+              )}`
+            )
+          }
```
[the remark this answers is not in the document]
Comment (stephenwf) [reply src/components/Search/SearchForm.js:63]: Styles looking good, the search is still a bit unstable when typing quickly. Also found a bug with this query for some reason: https://deploy-preview-94--delft-static-site-generator.netlify.app/en/search/?keywords=e not sure what's caused that!

File: src/components/Search/SearchForm.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  }, [val, delay]);
+
+  return debouncedValue;
+};
+
+const SearchForm = ({ pageLanguage, query, showTitle }) => {
+  const [value, setValue] = useState(query);
+  const debouncedSearchTerm = useDebounce(value, 500);
+
+  useEffect(
```
Comment (ionab) [inline src/components/Search/SearchForm.js:42]: @stephenwf Think my last commit has deleted your comments from here. I've added a useDebounce as suggested and it seems to be working a bit better for me. Would you be able to check the build tomorrow?

File: src/components/Search/SearchForm.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ionab) [reply src/components/Search/SearchForm.js]: Good find, how weird! Doesn't happen on the dev server.. Think I've pushed up a fix for this!

File: src/components/Search/SearchForm.js (hunk 5 above)
Comment (stephenwf) [reply src/components/Search/SearchForm.js:42]: Ah brilliant, yeh I'll take another look 👍

**Pull request merged**